File: TicketService/src/main/java/com/btlcnpm/TicketService/model/Ticket.java
package com.btlcnpm.TicketService.model;

import com.google.cloud.Timestamp;
import lombok.Data;

@Data
public class Ticket {

    private String ticketId;
    private String bookingId;
    private String userId;
    private String qrCodeData; // ticketId
    private String status;     // "VALID", "USED", "EXPIRED"
    private String movieTitle;
    private String theaterName;
    private String roomName;
    private Timestamp showtimeStartTime;
}

File: AndroidApp/app/src/main/java/com/btlcnpm/androidapp/data/model/Ticket.kt
package com.btlcnpm.androidapp.data.model

// Model này phải khớp với Ticket.java của TicketService
data class Ticket(
    val ticketId: String,
    val bookingId: String,
    val userId: String,
    val qrCodeData: String, // Đây chính là ticketId
    val status: String,     // "VALID", "USED", "EXPIRED"
    val movieTitle: String,
    val theaterName: String,
    val roomName: String,
    val showtimeStartTime: FirestoreTimestamp? // Dùng lại model FirestoreTimestamp đã tạo
)